Flaky test: TestRequestProfileInfo

[TestMessengerStoreNodeRequestSuite/TestRequestProfileInfo: 10.0% (2 of 20 failed)](https://ci.status.im/job/status-go/job/tests-nightly/142/Reports/)

```
/home/jenkins/workspace/status-go/tests-nightly/protocol/messenger_storenode_request_test.go:586
        	Error:      	Expected value not to be nil.
        	Test:       	TestMessengerStoreNodeRequestSuite/TestRequestProfileInfo
    --- FAIL: TestMessengerStoreNodeRequestSuite/TestRequestProfileInfo (2.35s)
```

https://ci.status.im/job/status-go/job/tests-nightly/142/testReport/github/com_status-im_status-go_protocol/TestMessengerStoreNodeRequestSuite_TestRequestProfileInfo/

--- a/protocol/messenger_store_node_request_manager.go
+++ b/protocol/messenger_store_node_request_manager.go
@@ -33,6 +33,17 @@ type storeNodeRequestID struct {
 	DataID      string               `json:"dataID"`
 }
 
+func (r *storeNodeRequestID) getCommunityID() string {
+	switch r.RequestType {
+	case storeNodeCommunityRequest:
+		return r.DataID
+	case storeNodeShardRequest:
+		return strings.TrimSuffix(r.DataID, transport.CommunityShardInfoTopicPrefix())
+	default:
+		return ""
+	}
+}
+
 type StoreNodeRequestManager struct {
 	messenger *Messenger
 	logger    *zap.Logger
@@ -502,24 +513,29 @@ func (r *storeNodeRequest) routine() {
 		r.finalize()
 	}()
 
-	communityIDStr := strings.TrimSuffix(r.requestID.DataID, transport.CommunityShardInfoTopicPrefix())
-	if !r.manager.messenger.communityStorenodes.HasStorenodeSetup(communityIDStr) && !r.manager.messenger.waitForAvailableStoreNode(storeNodeAvailableTimeout) {
-		r.result.err = fmt.Errorf("store node is not available")
-		return
+	communityID := r.requestID.getCommunityID()
+
+	if r.requestID.RequestType != storeNodeCommunityRequest || !r.manager.messenger.communityStorenodes.HasStorenodeSetup(communityID) {
+		if !r.manager.messenger.waitForAvailableStoreNode(storeNodeAvailableTimeout) {
+			r.result.err = fmt.Errorf("store node is not available")
+			return
+		}
 	}
-	ms := r.manager.messenger.getActiveMailserver(communityIDStr)
+
+	storeNode := r.manager.messenger.getActiveMailserver(communityID)
+
 	// Check if community already exists locally and get Clock.
-
-	localCommunity, _ := r.manager.messenger.communitiesManager.GetByIDString(r.requestID.DataID)
-
-	if localCommunity != nil {
-		r.minimumDataClock = localCommunity.Clock()
+	if r.requestID.RequestType == storeNodeCommunityRequest {
+		localCommunity, _ := r.manager.messenger.communitiesManager.GetByIDString(communityID)
+		if localCommunity != nil {
+			r.minimumDataClock = localCommunity.Clock()
+		}
 	}
 
 	// Start store node request
 	from, to := r.manager.messenger.calculateMailserverTimeBounds(oneMonthDuration)
 
-	_, err := r.manager.messenger.performMailserverRequest(ms, func(ms mailservers.Mailserver) (*MessengerResponse, error) {
+	_, err := r.manager.messenger.performMailserverRequest(storeNode, func(ms mailservers.Mailserver) (*MessengerResponse, error) {
 		batch := MailserverBatch{
 			From:        from,
 			To:          to,

--- a/protocol/messenger_storenode_request_test.go
+++ b/protocol/messenger_storenode_request_test.go
@@ -583,6 +583,7 @@ func (s *MessengerStoreNodeRequestSuite) TestRequestProfileInfo() {
 	s.Require().NoError(err)
 
 	s.createBob()
+	s.waitForAvailableStoreNode(s.bob)
 	s.fetchProfile(s.bob, s.owner.selfContact.ID, s.owner.selfContact)
 }
 

